src/Nethermind/Nethermind.Consensus/Processing/BranchProcessor.cs

```diff
@@ -51,46 +51,40 @@ public event EventHandler<TxProcessedEventArgs> TransactionProcessed
         remove => blockProcessor.TransactionProcessed -= value;
     }
 
-    private void InitBranch(BlockHeader? baseBlock, bool incrementReorgMetric = true)
-    {
-        /* Please note that we do not reset the state if branch state root is null.
-           That said, I do not remember in what cases we receive null here.*/
-        if (baseBlock is not null && _stateProvider.StateRoot != baseBlock.StateRoot)
-        {
-            /* Discarding the other branch data - chain reorganization.
-               We cannot use cached values any more because they may have been written
-               by blocks that are being reorganized out.*/
-
-            if (incrementReorgMetric)
-                Metrics.Reorganizations++;
```

Comment:
Reorg metric removed from this codepath as as the worldstate does not store nor allow getting stateroot outsite of it's scope.

Also, doesn't this mean this get incremented on all RPC call before?